RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -2968,6 +3375,28 @@ double PFAlgo::nSigmaHCAL(double clusterEnergyHCAL, double eta) const {
   return nS;
 }
 
+double PFAlgo::hfEnergyResolution(double clusterEnergyHF) const {
+  // Add a protection
+  if (clusterEnergyHF < 1.)
+    clusterEnergyHF = 1.;
+
+  double resol =
+      sqrt(pow(resolHF_a_, 2) / clusterEnergyHF + pow(resolHF_b_, 2) + pow(resolHF_c_, 2) / pow(clusterEnergyHF, 2));
```

Comment:
please save squares of resolHF_\*_ parameters to avoid unnecessary recomputations